apps/api/src/domain/models/user-management/user/commands/register-user.command.integration.spec.ts

```diff
@@ -0,0 +1,240 @@
+        name: new FullName({
+            firstName: 'Pete',
+            lastName: 'DeRepeat',
+        }),
+    }),
+};
+
+const existingUser = new CoscradUser(existingUserDto);
+
+const buildInvalidFSA: InvalidFSAFactoryFunction<RegisterUser> = (
+    id: AggregateId,
+    payloadOverrides: Partial<Record<keyof RegisterUser, unknown>> = {}
+): FluxStandardAction<DTO<RegisterUser>> => ({
+    type: commandType,
+    payload: {
+        ...buildValidCommandFSA(id).payload,
+        ...(payloadOverrides as Partial<RegisterUser>),
+    },
+});
+
+const initialState = buildEmptyInMemorySnapshot();
+
+describe('RegisterUser', () => {
+    let testRepositoryProvider: TestRepositoryProvider;
+
+    let commandHandlerService: CommandHandlerService;
+
+    let arangoConnectionProvider: ArangoConnectionProvider;
+
+    let idManager: IIdManager;
+
+    let commandAssertionDependencies: CommandAssertionDependencies;
+
+    beforeAll(async () => {
+        ({ testRepositoryProvider, commandHandlerService, idManager, arangoConnectionProvider } =
+            await setUpIntegrationTest({
+                ARANGO_DB_NAME: generateRandomTestDatabaseName(),
+            }));
+
+        commandHandlerService.registerHandler(
+            commandType,
+            new RegisterUserCommandHandler(testRepositoryProvider, idManager)
+        );
+
+        commandAssertionDependencies = {
+            testRepositoryProvider,
+            idManager,
+            commandHandlerService,
+        };
+    });
+
+    afterAll(async () => {
+        await arangoConnectionProvider.dropDatabaseIfExists();
+    });
+
+    beforeEach(async () => {
+        await testRepositoryProvider.testSetup();
+    });
+
+    afterEach(async () => {
+        await testRepositoryProvider.testTeardown();
+    });
+
+    describe('when the command is valid', () => {
+        it('should succeed', async () => {
+            await assertCreateCommandSuccess(commandAssertionDependencies, {
+                buildValidCommandFSA,
+                initialState,
+                checkStateOnSuccess: async ({ id }: RegisterUser) => {
+                    const userSearchResult = await testRepositoryProvider
+                        .getUserRepository()
+                        .fetchById(id);
+
+                    expect(userSearchResult).toBeInstanceOf(CoscradUser);
+                },
+            });
+        });
+    });
+
+    describe('when the command is invalid', () => {
+        describe('when the external state is invalid', () => {
+            describe('when there is already a user with the given id', () => {
+                it('should fail', async () => {
+                    const newId = await idManager.generate();
+
+                    await testRepositoryProvider.addFullSnapshot(
+                        buildInMemorySnapshot({ users: [existingUser.clone({ id: newId })] })
+                    );
+
+                    const validFSA = buildValidCommandFSA(newId);
```

Comment:
Rename this variable and add a comment that explains that you are invalidating the command by invalidating existing state.